Logout issue 451

RueHaile opened the pull request:
Resolved issue: [451_ LSF](https://github.com/BCStudentSoftwareDevTeam/lsf/issues/451)

_Fixed:_

- The log out button on LSF student side has been fixed so that they will be logged out successfully and also  communicated about it by removing the remove error message and showing a shibboleth logged out message on a new web page.

_Issue:_

 -Logout page doesn’t log you out and only redirects to basic HTML template (On student side). 
- It gave a maximum recursion depth reached error, on the side of development and an unexpected error occurred on the production side when the user attempts to log out. 
-The user was not informed that they have successfully checked out only that there has been:
“Unexpected error has occurred, an administrator has been notified and sorry for the problem page”.

**_Testing the issue :Development error_** 
![image](https://github.com/user-attachments/assets/abd21068-fa86-4258-8d76-b6c2a0bd72f1)

**_Testing the issue: production/published side_** 
![image](https://github.com/user-attachments/assets/f0d34145-e4fc-4e50-8882-a4d9a69e2cc8) 
 
**_Currently tested to output/ resolved :_**  
![image](https://github.com/user-attachments/assets/32e8c5e4-1e35-45cc-b132-5fefdf9cb298)

**_When the user is logged out, they are redirected to the Duo login page:-_**
![image](https://github.com/user-attachments/assets/5d7c766b-e56e-41b4-8122-b3cc93fc2565)
**_If the user presses back on the web browser, a stale request will be rendered._** 
![image](https://github.com/user-attachments/assets/166ffe43-f491-4bb6-8823-b10e6e25aa94)

 


 
Test: 
•	Normally, students the way that we can get the labor form website if we are provided with a link from a supervisor.
•	Check out branch:   logout_issue_45.
•	Once on the page, there is a sidebar with a logout button on it. 
•	Click the log out and followed by a “logged-out operation is complete page.
•	The user has successfully logged out; they can exit the page.

RueHaile committed:
Fixed the infinite recall problem and added a landing page of labor program. It is still not fixed we need to find a better landing page. it is a work in progress.

RueHaile committed:
- After Logging out the user and clearing the session successfully, a Shibboleth logged out page is renderd.
-The logout issue on the student side has been resolve from rendering an infinite cycle error to a Shibboleth logged out page.

bakobagassas commented:
Everything looks fine

adjagbodjouf commented:
After logging out, the successful logout page loads, indicating that the user has been logged out. But after this page shows up, you can click the back arrow everything still works fine, like the user still hasn't logged out. Even after refreshing, everything appears to still be working when it shouldn't. This may just be because this is all running locally, but if not, then it may be something you need to fix.



**Responses:**
- The back arrow won't be rendering the web page anymore without the user logging in.
-  The back button will not work when they are logged out.
**_This issue has been resolved._**

adjagbodjouf commented:
This would look better centered on the page instead of being left all the way at the top.

![image](https://github.com/user-attachments/assets/02b76047-b431-44f2-8f00-4640d3247f01)

The text might also look better centered in the box.

RueHaile committed:
-A Shibboleth log out has been added.
-The redirect to the logout in the main route is added.
-In the HTML, prevent back is added in order to remove the history and as a result prevent the back button from working.

RueHaile commented:
> This would look better centered on the page instead of being left all the way at the top.
> 
> ![image](https://private-user-images.githubusercontent.com/156280375/462141683-02b76047-b431-44f2-8f00-4640d3247f01.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTE5ODg5ODcsIm5iZiI6MTc1MTk4ODY4NywicGF0aCI6Ii8xNTYyODAzNzUvNDYyMTQxNjgzLTAyYjc2MDQ3LWI0MzEtNDRmMi04ZjAwLTQ2NDBkMzI0N2YwMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcwOFQxNTMxMjdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NmE3MGFkODcyYmRjODBmN2MwOWZjNmY1MzljODRiNGI4MzU2ODc4ZWIzZDg3ZjliZjExZjU1YTRhZTA2NWZkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.scbYnRv0dkeOFDdSpRCFWzFydqMf1hWuYQLJvmAtmsc)
> 
> The text might also look better centered in the box.

**Response:** 
- This is no longer an issue because we are rendering a different page.
_**This issue has been resolved.**_

BesherKitaz commented:
I got this error when trying to open the site on the test database when I try to open the website as a student (I set the user to 'kitazb' in secret_config.yaml, which is a student username):


<img width="1892" height="940" alt="image" src="https://github.com/user-attachments/assets/5e765a30-09d4-4830-8f0d-627f14b3d4b3" />

When I set the username back to "heggens" to view the supervisor portal:

<img width="1914" height="935" alt="image" src="https://github.com/user-attachments/assets/26715b47-e1b3-48f9-b526-6ee2214d4df8" />

And then when I clicked "back," it took me to my profile in lsf production site:

<img width="1913" height="927" alt="image" src="https://github.com/user-attachments/assets/70a926b9-42b3-4255-96a7-c0035ce94bee" />

RESPONSE: 
Clearing the cache should resolve the issue.

BrianRamsay reviewed (changes requested):
This fixes the recursion error, but doesn't actually log them out. We need to figure out what we need to configure so that going to `https://secure.berea.edu/Shibboleth.sso/Logout` actually ends their duo/shibboleth session.

If we can't figure that part out, we should just remove the Logout link altogether.

BrianRamsay commented:
I implemented just the logout fix in commit 5d2fca22. Had to find the right url redirect and make a config change in `/etc/shibboleth/shibboleth2.xml` in production.
